tests/src/test/scala/com/nvidia/spark/rapids/tests/udf/scala/URLDecode.scala

```diff
@@ -40,11 +40,12 @@ class URLDecode extends Function[String, String] with RapidsUDF with Serializabl
   }
 
   /** Columnar implementation that runs on the GPU */
-  override def evaluateColumnar(args: ColumnVector*): ColumnVector = {
+  override def evaluateColumnar(numRows: Int, args: ColumnVector*): ColumnVector = {
     // The CPU implementation takes a single string argument, so similarly
     // there should only be one column argument of type STRING.
     require(args.length == 1, s"Unexpected argument count: ${args.length}")
     val input = args.head
+    require(numRows == input.getRowCount, s"Expected $numRows rows, received ${input.getRowCount}")
```

Comment:
nit: Why need to check the numRows's correctness, but no where to use it in URLDecode case?
It should be ok to skip this line, if URLDecode won't use numRows.
I think in most case of using RapidsUDF, we just need to add the numRows parameter and the logic should be as same as before. Only the case what you mentioned in the issue, which numRows has number but no data in the input ColumnVector, we should use numRows to do something like creating a scalar.
Am I right？

Reply:
Given this UDF is part of a unit test for RAPIDS accelerated UDFs, the assert is warranted.  If we ever pass an incorrect value here, that would be quite bad.  The point of checking it here is to validate the framework is doing the right thing from the user's perspective (i.e.: the one implementing the RAPIDS accelerated UDF code).